ethereum/test/walletBackend.go

```diff
@@ -0,0 +1,95 @@
+package test
+
+import (
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+
+	ethwallet "perun.network/go-perun/backend/ethereum/wallet"
+	"perun.network/go-perun/wallet"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/direct-state-transfer/dst-go/ethereum"
+	"github.com/direct-state-transfer/dst-go/identity"
+)
+
+const (
+	weakScryptN = 2
+	weakScryptP = 1
+)
+
+type WalletBackendSetup struct {
+	WalletBackend *ethereum.WalletBackend
+	Keystore      *keystore.KeyStore
+	KeystorePath  string
+	User          *identity.User
+}
+
+func NewUserSetup(t *testing.T, rng *rand.Rand, tempDir string) (WalletBackendSetup, func()) {
+	ws, walletCleanup := NewWalletSetup(t, rng, tempDir, 3)
+	user := &identity.User{
+		Peer: identity.Peer{
+			Alias:   "test-user",
+			PerunID: ws.Accounts[0],
+		},
+		OnChainAcc:     ws.Accounts[1],
+		OnChainWallet:  ws.Wallet,
+		OffChainAcc:    ws.Accounts[2],
+		OffChainWallet: ws.Wallet,
+	}
+	cleanup := func() { walletCleanup() }
+
+	return WalletBackendSetup{
+		WalletBackend: ws.WalletBackend,
+		Keystore:      ws.Keystore,
+		KeystorePath:  ws.KeystorePath,
+		User:          user,
+	}, cleanup
+}
+
+type WalletSetup struct {
+	WalletBackend *ethereum.WalletBackend
+	KeystorePath  string
+	Keystore      *keystore.KeyStore
+	Wallet        wallet.Wallet
+	Accounts      []wallet.Account
+}
+
+// NewWallet initializes a wallet with n accounts. Empty password string and weak encrytion parameters are used.
```

Comment:
```suggestion
// NewWalletSetup initializes a wallet with n accounts. Empty password string and weak encrytion parameters are used.
```

Reply:
Updated.

Reply:
I think you `git` ate that line. It is still "`NewWallet`".

Reply:
yeah.. I it is updated now.